### ISSUE TITLE
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

### ISSUE THREAD

**Issue by VeigarJia** [DataCanvasIO/YLearn#25]
In section 4.A. in `case_study_bank.ipynb`, when I run `why.fit(train_data,outcome)`, an error has occurred:
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)

It only occurs when variable `treatment` was not assigned. How can I fix it?

**Comment by VeigarJia** [comment]
1. macOS Big Sur 11.6.1
2. Python 3.8.12
3. ylearn 0.1.1
4. pip list
```
Package                           Version
--------------------------------- -------------------
absl-py                           0.15.0
aiohttp                           3.8.1
aiosignal                         1.2.0
anaconda-client                   1.9.0
anaconda-navigator                2.1.1
analytics-python                  1.2.9
anyio                             3.4.0
appnope                           0.1.2
APScheduler                       3.7.0
argon2-cffi                       20.1.0
asttokens                         2.0.5
astunparse                        1.6.3
async-generator                   1.10
async-timeout                     4.0.2
attrs                             21.2.0
autokeras                         1.0.16.post1
autopep8                          1.6.0
Babel                             2.9.1
backcall                          0.2.0
bamboolib                         1.26.0
beautifulsoup4                    4.10.0
bleach                            4.0.0
Bottleneck                        1.3.2
bqplot                            0.12.31
brotlipy                          0.7.0
cachetools                        4.2.4
causal-learn                      0.1.2.4
causalml                          0.12.3
causalnex                         0.11.0
certifi                           2021.10.8
cffi                              1.14.6
chardet                           4.0.0
charset-normalizer                2.0.4
click                             8.0.3
cloudpickle                       2.0.0
clyent                            1.2.2
conda                             4.12.0
conda-build                       3.21.4
conda-content-trust               0+unknown
conda-package-handling            1.7.3
conda-repo-cli                    1.0.4
conda-token                       0.3.0
conda-verify                      3.1.1
cryptography                      3.4.8
cycler                            0.11.0
Cython                            0.29.30
d2l                               0.17.5
daal4py                           2021.4.0
dataframe-image                   0.1.1
DBUtils                           1.3
deap                              1.3.1
debugpy                           1.5.1
decorator                         5.1.0
defusedxml                        0.7.1
Deprecated                        1.2.13
dill                              0.3.5.1
dowhy                             0.7
entrypoints                       0.3
et-xmlfile                        1.1.0
executing                         0.8.3
expiringdict                      1.2.1
fasttext                          0.9.2
filelock                          3.4.0
Flask                             2.1.1
flatbuffers                       1.12
fonttools                         4.25.0
frozenlist                        1.3.0
future                            0.18.2
gast                              0.4.0
gensim                            4.2.0
glob2                             0.7
gnureadline                       8.0.0
google-auth                       2.3.3
google-auth-oauthlib              0.4.6
google-pasta                      0.2.0
graphviz                          0.20
grpcio                            1.34.1
h5py                              3.1.0
handcalcs                         1.4.1
huggingface-hub                   0.0.8
hyperopt                          0.2.7
idna                              2.10
imbalanced-learn                  0.8.1
imgkit                            1.2.2
importlib-metadata                4.8.2
importlib-resources               5.4.0
iniconfig                         1.1.1
innerscope                        0.5.0
ipykernel                         6.6.0
ipympl                            0.7.0
ipyslickgrid                      0.0.3
ipython                           8.4.0
ipython-genutils                  0.2.0
ipywidgets                        7.6.5
itables                           0.3.1
itsdangerous                      2.1.2
jedi                              0.18.1
jieba                             0.42.1
Jinja2                            3.0.2
joblib                            1.1.0
json5                             0.9.6
jsonschema                        4.3.1
jupyter                           1.0.0
jupyter-client                    7.1.0
jupyter-console                   6.4.0
jupyter-contrib-core              0.3.3
jupyter-contrib-nbextensions      0.5.1
jupyter-core                      4.9.1
jupyter-highlight-selected-word   0.2.0
jupyter-latex-envs                1.4.6
jupyter-nbextensions-configurator 0.4.1
jupyter-pip                       0.3.1
jupyter-server                    1.13.1
jupyterlab                        3.2.5
jupyterlab-pygments               0.1.2
jupyterlab-server                 2.9.0
jupyterlab-widgets                1.0.0
jupyterthemes                     0.20.0
keras                             2.7.0
keras-nightly                     2.5.0.dev2021032900
Keras-Preprocessing               1.1.2
keras-tuner                       1.1.0
kiwisolver                        1.3.1
kt-legacy                         1.0.4
lesscpy                           0.15.0
libarchive-c                      2.9
lightgbm                          3.2.1
llvmlite                          0.37.0
loguru                            0.6.0
ltp                               4.1.5.post2
lxml                              4.6.4
Markdown                          3.3.6
MarkupSafe                        2.0.1
matplotlib                        3.5.1
matplotlib-inline                 0.1.3
md-mermaid                        0.1.1
mistune                           0.8.4
mkl-fft                           1.3.1
mkl-random                        1.2.2
mkl-service                       2.4.0
mlbox                             0.5.1
mock                              4.0.3
more-itertools                    8.12.0
mpmath                            1.2.1
multidict                         6.0.2
multiprocess                      0.70.13
munkres                           1.1.4
navigator-updater                 0.2.1
nb-conda                          2.2.1
nb-conda-kernels                  2.3.1
nb-mermaid                        0.1.0
nbclassic                         0.3.4
nbclient                          0.5.3
nbconvert                         5.6.1
nbformat                          5.1.3
nest-asyncio                      1.5.4
networkx                          2.6.3
nltk                              3.6.7
notebook                          6.4.6
numba                             0.54.1
numexpr                           2.7.3
numpy                             1.20.3
oauthlib                          3.1.1
olefile                           0.46
openpyxl                          3.0.9
opt-einsum                        3.3.0
packaging                         21.3
pandas                            1.3.3
pandocfilters                     1.4.3
parso                             0.8.3
pathos                            0.2.9
patsy                             0.5.2
pdfkit                            1.0.0
pexpect                           4.8.0
pgmpy                             0.1.18
pickleshare                       0.7.5
Pillow                            8.4.0
pip                               21.2.4
pkginfo                           1.7.1
plotly                            4.14.3
pluggy                            1.0.0
ply                               3.11
pox                               0.3.1
ppft                              1.7.6.5
ppscore                           1.2.0
prettytable                       2.4.0
progressbar2                      4.0.0
prometheus-client                 0.12.0
prompt-toolkit                    3.0.24
protobuf                          3.19.1
psutil                            5.8.0
ptyprocess                        0.7.0
pure-eval                         0.2.2
py                                1.11.0
py4j                              0.10.9.3
pyasn1                            0.4.8
pyasn1-modules                    0.2.8
pybind11                          2.9.2
pycodestyle                       2.8.0
pycosat                           0.6.3
pycparser                         2.21
pydot                             1.4.2
pydotplus                         2.0.2
pyecharts                         1.9.1
pygam                             0.8.0
Pygments                          2.10.0
pygtrie                           2.4.2
PyJWT                             2.1.0
PyMySQL                           1.0.2
pyOpenSSL                         21.0.0
pyparsing                         3.0.6
PyQt5                             5.12.3
PyQt5_sip                         4.19.18
PyQtChart                         5.12
PyQtWebEngine                     5.12.1
pyro-api                          0.1.2
pyro-ppl                          1.8.1
pyrsistent                        0.18.0
PySocks                           1.7.1
pytest                            7.1.1
python-dateutil                   2.8.2
python-utils                      3.3.3
pytz                              2021.3
PyYAML                            6.0
pyzmq                             22.3.0
qtconsole                         5.1.1
QtPy                              1.10.0
redis                             4.3.3
regex                             2021.11.10
requests                          2.25.1
requests-oauthlib                 1.3.0
retrying                          1.3.3
rise                              5.7.1
rsa                               4.8
ruamel-yaml-conda                 0.15.100
runipy                            0.1.5
sacremoses                        0.0.53
schedule                          1.1.0
scikit-learn                      0.24.2
scikit-learn-intelex              2021.20211201.94539
scipy                             1.7.1
seaborn                           0.10.1
Send2Trash                        1.8.0
setuptools                        60.0.3
shap                              0.40.0
simplejson                        3.17.6
six                               1.16.0
sklearn-contrib-py-earth          0.1.0
sklearn-pandas                    2.2.0
slicer                            0.0.7
smart-open                        6.0.0
sniffio                           1.2.0
soupsieve                         2.3.1
stack-data                        0.3.0
statsmodels                       0.13.1
stopit                            1.1.2
sympy                             1.9
tables                            3.6.1
tabulate                          0.8.9
tensorboard                       2.7.0
tensorboard-data-server           0.6.1
tensorboard-plugin-wit            1.8.0
tensorflow                        2.5.2
tensorflow-estimator              2.5.0
termcolor                         1.1.0
terminado                         0.9.4
testpath                          0.5.0
Theano                            1.0.5
threadpoolctl                     2.2.0
tokenizers                        0.10.3
toml                              0.10.2
tomli                             1.2.2
toolz                             0.11.2
torch                             1.11.0
torchaudio                        0.11.0
torchvision                       0.12.0
tornado                           6.1
TPOT                              0.11.7
tqdm                              4.62.3
traitlets                         5.1.1
traittypes                        0.2.1
transformers                      4.7.0
typing-extensions                 3.7.4.3
tzlocal                           2.0.0
ujson                             4.0.2
update-checker                    0.18.0
urllib3                           1.26.7
ViTables                          3.0.2
wcwidth                           0.2.5
webencodings                      0.5.1
websocket-client                  1.2.3
Werkzeug                          2.0.2
wheel                             0.37.0
widgetsnbextension                3.5.1
wkhtmltopdf                       0.2
wordcloud                         1.8.2.2
wrapt                             1.12.1
xgboost                           1.5.0
xlrd                              1.2.0
XlsxWriter                        3.0.3
xmltodict                         0.12.0
xpinyin                           0.7.6
yarl                              1.7.2
ylearn                            0.1.1
zipp                              3.6.0
```
5. The full error stack trace:
Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)
![企业微信20220715-115645](https://user-images.githubusercontent.com/15810776/179147310-a075850c-0685-4d2a-88bd-01f92d9a1542.png)

**Comment by lixfz** [comment]
@VeigarJia  Thanks very much for the information.

YLearn has some conflicts with lightgbm on macOS, we will fix this issue in next release.
As a workaround way on macOS, install YLearn in python env without lightgbm please.

**Comment by lixfz** [comment]
This issue was fixed in version 0.1.2, upgrade ylearn to resolve it.
@VeigarJia
